codec-http/src/main/java/io/netty/handler/codec/http/websocketx/extensions/compression/DeflateDecoder.java

```diff
@@ -136,8 +136,12 @@ private ByteBuf decompressContent(ChannelHandlerContext ctx, WebSocketFrame msg)
         // Correctly handle empty frames
         // See https://github.com/netty/netty/issues/4348
         if (!emptyDeflateBlock && readable && compositeDecompressedContent.numComponents() <= 0) {
-            compositeDecompressedContent.release();
-            throw new CodecException("cannot read uncompressed buffer");
+            // Sometimes after fragmentation the last frame
+            // May contain trash that doesn't affect decompression
+            if (!(msg instanceof ContinuationWebSocketFrame)) {
```

Comment:
hmm... I am wondering if it is really a good idea to ignore stuff as long as its `ContinuationWebSocketFrame `.

Reply:
Maybe not for all `ContinuationWebSocketFrame` fragments only for final. In most of implementation that i saw thety don't worry at all about decompressed payload is empty or not.

Reply:
I am just wondering why we would receive trash... this does sounds very odd to me. What I am missing ?

Reply:
@normanmaurer check test please, last frame may contain bytes which no longer affect the decompression result so `inflater.inflate(...)` return 0.

Reply:
@amizurov even after looking at the test I don't understand it... Can you please expand the comment above to explain why it contains trash ? I am not even sure what "trash" means in this context

Reply:
Also maybe you could also link to another compression implementation that does the same. Maybe this clears things up.

Reply:
@normanmaurer sorry if i confused you maybe "trash" inappropriate word in this context, one of implementation you can look here [undertow](https://github.com/undertow-io/undertow/blob/b6bd4d2e6611f135e740d714cee2e811e9815ef8/core/src/main/java/io/undertow/websockets/extensions/PerMessageDeflateFunction.java#L165), the `output` can be empty buffer and this is ok in some cases.

Reply:
@amizurov I see... yes maybe just fix the content to replace trash with "extra data" or something like that ?